main.go

```diff
@@ -168,6 +174,62 @@ func wrapMain() error {
 		"namespace": vargs.Namespace,
 	}
 
+	if !vargs.DryRun && vargs.EndpointsSchema != "" {
+		// get latest deployed version. Newest first
+		err = runner.Run(vargs.GCloudCmd, "gcloud", "alpha", "endpoints", "config", "list", "--service=", vargs.EndpointsService, "--format", "'value(id)'", "--limit", "1")
+		if err != nil {
+			return fmt.Errorf("Error: %s\n", err)
+		}
+		serviceVersion := runner.Output()
+		swaggerVersion := []byte{}
+		var version struct {
+			Info struct {
+				Version string `json:"version"`
+			} `json:"info"`
+		}
+
+		// schema has been deployed previously
+		if len(serviceVersion) != 0 {
+			// get deployed version of swagger file
+			err = runner.Run(vargs.GCloudCmd, "gcloud", "alpha", "endpoints", "config", "describe", string(serviceVersion), "--service=", vargs.EndpointsService, "--format", "'value(apis.version)'")
+			if err != nil {
+				return fmt.Errorf("Error: %s\n", err)
+			}
+			swaggerVersion = runner.Output()
+
+			// read version from schema
+			f, err := os.Open(vargs.EndpointsSchema)
+			if err != nil {
+				return fmt.Errorf("Error: %s\n", err)
+			}
+
+			fi, err := f.Stat()
+			if err != nil {
+				return fmt.Errorf("Error: %s\n", err)
+			}
+			err = yaml.NewYAMLOrJSONDecoder(f, int(fi.Size())).Decode(&version)
+			if err != nil {
+				return fmt.Errorf("Error: %s\n", err)
+			}
+			f.Close()
+		}
+		data["endpoints_service_name"] = vargs.EndpointsService
+
+		if len(serviceVersion) == 0 || string(swaggerVersion) != version.Info.Version {
```

Comment:
Is this where you are comparing the deployed version with the version specified in the file using the semver of the openAPI spec?

Reply:
yes